### PR TITLE
Update to HelloWorld.cs

### PR DESCRIPTION
**Pull request by ryaneliseislalom**
Updated corefxlab/demos/CoreClrConsoleApplications/HelloWorld/HelloWorld.cs to include a function for OS detection while still allowing args[0] to be used for overriding and showing various OS logos. No break to existing behavior.

Also note, that proper OS detection would use Diagnostics.Environment, which is not yet implemented in CoreFX. This has been noted in the detection method comments.

### CODE REVIEW

File: HelloWorld.cs [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+                    Console.BackgroundColor = colors[row * 2 + col];
+                    Console.ForegroundColor = colors[row * 2 + col];
+                    for (int j = 0; j < squareSize; j++) Console.Write('@');
+                    Console.ResetColor();
+                }
+            }
+        }
+        Console.WriteLine();
+    }
+
```
Comment (KrzysztofCwalina) [inline HelloWorld.cs]: Why all these show as new? Did you change anything in the Draw* methods?

Comment (ryaneliseislalom) [inline HelloWorld.cs]: I didn’t see a place to create a feature branch, but did see an option to submit a file in the fork dialog. Please advise if change is needed. Summary is addition of new function, and minor modification of conditionals in Main. No other changes.

File: HelloWorld.cs [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+            }
+
+        }
+
+        Console.ResetColor();
+        Console.WriteLine();
+    }
+
+    //HACK: Needed until CoreFX implements Environment.OSVersion
+    private static string DetectOS()
```
Comment (ryaneliseislalom) [inline HelloWorld.cs]: This method is new

File: HelloWorld.cs [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+
+using System;
+using System.Collections;
+using System.Diagnostics;
+
+internal class Program
+{
+    private static void Main(string[] args)
+    {
+        string osDetect = DetectOS();
```
Comment (ryaneliseislalom) [inline HelloWorld.cs]: This detection behavior is new

File: HelloWorld.cs [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+        {
+            string shellValue = proc.StartInfo.Environment["SHELL"].ToLowerInvariant();
+
+            if (shellValue.EndsWith("/tcsh") || shellValue.EndsWith("/sh"))
+            {
+                detectionResult = "freebsd";
+            } else {
+                detectionResult = "linux";
+            }
+        } else if (proc.StartInfo.Environment.ContainsKey("COMSPEC"))
```
Comment (bartonjs) [inline HelloWorld.cs]: The bracing on the else/else-ifs should be
```
if (condition)
{
    //body;
}
else if (condition2)
{
    //body2;
}
else
{
    //body3;
}
```

https://github.com/dotnet/corefx/blob/master/Documentation/coding-style.md (rule 1)

**Pull request closed without merging**